gobblin-runtime/src/main/java/gobblin/refactor/FSJobCatalog.java

```diff
@@ -0,0 +1,85 @@
+package gobblin.refactor;
+
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gobblin.configuration.ConfigurationKeys;
+
+
+/**
+ * The job Catalog for file system to persist the job configuration information.
+ * For stateful storage like file system,
+ */
+public class FSJobCatalog implements JobCatalog {
+
+  private Path jobConfDirPath;
+  private final List<JobCatalogListener> listeners = new CopyOnWriteArrayList<>();
+  private HashMap<URI, JobSpec> persistedJobs = new HashMap<>();
+  private static final Logger LOGGER = LoggerFactory.getLogger(FSJobCatalog.class);
+  private Configuration conf = new Configuration();
+  private FileSystem fs;
+
+  /**
+   * Initialize the JobCatalog, fetch all jobs in jobConfDirPath.
+   * @param jobConfDirPath
+   * @throws Exception
+   */
+  public FSJobCatalog(Path jobConfDirPath)
+      throws Exception {
```

Comment:
Please avoid throwing a generic Exception as it does not tell anything about the nature of exceptions that can be thrown. It defeats the purpose of using checked exceptions. Either use specific exception types or just let the implementation throw RuntimeException.